src/mailroom.py

```diff
@@ -0,0 +1,155 @@
+                             '(Q) - Exit : ').lower()
+    return user_menu_choice
+
+
+def initial_prompt_validator(user_menu_choice):
+    """Validate user input from the initial_prompt."""
+    if user_menu_choice in (u'l', u'r'):
+        return user_menu_choice
+    elif user_menu_choice == 'q':
+        exit()
+    else:
+        print(u'Invalid Input')
+        return u'invalid input'
+
+
+def name_prompt():
+    """Prompt user for a name, or asks for list to print the names in dict."""
+    donor_name_input = input(u'Enter Donor Name ((L) - List of Donors '
+                             u'(Q) - return to Main Menu) : ').lower()
+    return donor_name_input
+
+
+def name_prompt_validator(donor_name_input):
+    """Validate name_prompt's input."""
+    if donor_name_input == u'q':
+        main_function()
+    else:
+        try:
+            if int(donor_name_input) or float(donor_name_input):
+                print(u'Invalid Input')
+                return u'invalid input'
+        except ValueError:
+            if donor_name_input != ' ':
+                return donor_name_input
+            else:
+                print(u'Invalid Input')
+                return u'invalid input'
+
+
+def amount_prompt():
+    """Prompt for donation amount."""
+    donation_amount = input(u'Enter Donation Amount  '
+                            u'((Q) - return to Main Menu) : ')
+    return donation_amount
+
+
+def amount_prompt_validator(donation_amount):
+    """Validate donation amount."""
+    if donation_amount == u'q':
+            main_function()
+    else:
+        try:
+            if int(donation_amount) or float(donation_amount):
+                return float(donation_amount)
+        except ValueError:
+            print(u'Invalid Input')
+            return u'invalid input'
+
+
+def update_dict(donor_name, donation_amount, dictionary):
+    """Check if name is in dict, add if not and add donation to the dict."""
+    dictionary.setdefault(donor_name, []).append(donation_amount)
+    return dictionary
+
+
+def generate_letter(donor_name, donation_amount):
+    """Generate thank you letter with name and new donation."""
+    print(u'Dear {0}, thank you for your donation of '
+          u'${1}.'.format(donor_name.capitalize(), donation_amount))
+    return (u'Dear {0}, thank you for your donation of '
+            u'${1}.'.format(donor_name.capitalize(), donation_amount))
+
+
+def do_math(dictionary):
+    """Find sum, no. of donations, and avg of donations for individuals."""
+    summary_list = []
+    for key in dictionary:
+        donor_math_list = []
+        value = dictionary[key]
+        total = sum(value)
+        count = len(value)
+        average = total / count
+        donor_math_list = [key, total, count, average]
+        summary_list.append(donor_math_list)
+    return sorted(summary_list, key=lambda x: x[1])
+
+
+def create_report(list_sorted):
+    """Print report of donors, with sum, count of, and avg of donations."""
```

Comment:
a problem we have is that this function is pretty much  un-testable as it is.  You could do some fancy stuff to get access to the results of calling `print`, but the fact that you do is problematic.  I would suggest that you make two functions, one that formats all your print lines, and a second that prints the lines it is given.  The second will be trivial and not require testing.  The first will be testable because it will return data.